app/models/piece.rb

```diff
@@ -32,12 +32,81 @@ def x_y_coords
     x_y_coordinates
   end
 
-  def is_obstructed?(x, y)
-    false
-  end
-
   def set_image
     color ? color_string = "white" : color_string = "black"
     self.image_name ||= "#{color_string}-#{piece_type.downcase}.png"
   end
+
+  def is_obstructed?(x, y)
+    dest_x = x
+    dest_y = y
+    state_x = x_position
+    state_y = y_position
+    delta_x = dest_x - state_x
+    delta_y = dest_y - state_y
+
+    if delta_y == 0 && delta_x > 0 && delta_x.abs > 1
+      # horizontal move where start is < destination and distance > 1
+      return self.game.pieces.where("y_position = ? AND x_position BETWEEN ? AND ?", state_y, (state_x + 1), (dest_x - 1)).any?
+    elsif delta_y == 0 && delta_x < 0 && delta_x.abs > 1
+      # horizontal move where start is > destination and distance > 1
+      return self.game.pieces.where("y_position = ? AND x_position BETWEEN ? AND ?", state_y, (dest_x + 1), (state_x - 1)).any?
+    elsif delta_y == 0 && delta_x.abs <= 1
+      # horizontal move to next square or delta 0
+      return false
+    elsif delta_x == 0 && delta_y > 0 && delta_y.abs > 1
+      # vertical move where start is < destination and distance > 1
+      return self.game.pieces.where("x_position = ? AND y_position BETWEEN ? AND ?", state_x, (state_y + 1), (dest_y - 1)).any?
+    elsif delta_x == 0 && delta_y < 0 && delta_y.abs > 1
+      # vertical move where start is > destination and distance > 1
+      return self.game.pieces.where("x_position = ? AND y_position BETWEEN ? AND ?", state_x, (dest_y + 1), (state_y - 1)).any?
+    elsif delta_x == 0 && delta_y.abs <= 1
+      # vertical move to next square or delta 0
+      return false
+    elsif delta_x == delta_y && delta_x > 0
+      # SE move, positive X, positive Y diagonal
+      steps = delta_x - 1
+      while steps != 0
```

Comment:
this is probably fine. just makes me a little nervous. i know with the logic it will be fine, just gets me nervous about infinite loops. i'd suggest guarding against it with `while steps >0`,  but i'm overly paranoid sometimes. just a thought.